src/storage/v2/inmemory/storage.cpp

```diff
@@ -2398,8 +2346,20 @@ std::vector<std::pair<LabelId, PropertyId>> InMemoryStorage::InMemoryAccessor::D
 }
 
 void InMemoryStorage::InMemoryAccessor::DropGraph() {
+  // STEP 1 of GC Cleanup
+  std::list<Gid> current_deleted_vertices;
+  std::list<Gid> current_deleted_edges;
+  GCRapidDeltaCleanup(current_deleted_vertices, current_deleted_edges);
```

Comment:
This now makes it clear that `DropGraph` needs different Delta cleanup. You know all vertexes and edges are being deleted. You don't need to collect a subset to remove, because all of them will be removed. With that in mind, you do not need to unlink any delta, you just have to prevent GC from navigating and cleaning those deltas by taking they away from GCs control.